### PR TITLE
Fetch users with 0 usage directly when fetching usages by coalescing

### PR DESCRIPTION
**Pull request by ianmcorvidae**
(and left-joining/matching on null)

This should mean that we get 0s when fetching batch usages, rather than only getting values for users with >0 usage.

### CODE REVIEW

**Comment by ianmcorvidae** [comment]
Hopefully, we could also remove https://github.com/cyverse-de/data-usage-api/blob/main/db/de.go#L120 after merging this (i.e. the older hack). But we should get this in first and be semi-confident it's working, probably.

File: db/de.go [hunk 1]
```diff
@@ -233,6 +233,10 @@ func (d *DEDatabase) GetUserInfo(context context.Context, username string) (*Use
 		return nil, errors.Wrap(err, "error getting user info")
 	}
 
+	if len(uis) < 1 {
+		return nil, errors.New("No rows returned for user")
+	}
```
Comment (ianmcorvidae) [inline db/de.go:139]: This caused a panic due to the `uis[0]` below, so I added a check.

**Comment by ianmcorvidae** [comment]
Heh, looking again, the older hack isn't even actually called anymore, meaning we can probably delete the entire function it's in, _and_ that we haven't been doing any 0 values as part of batches, which does explain some things.

**Comment by ianmcorvidae** [comment]
Reviews are still welcome here, but I think I'm going to go ahead and merge this; especially with my earlier 0-usage hack being routed around, I think we probably want this somewhat urgently if possible, so I'd like to test it some.

**Comment by slr71** [comment]
These changes look good to me! 👍

**Pull request merged**